constraints.pro

```diff
@@ -117,6 +117,20 @@ repo_name(RepoUrl, RepoName) :-
   RepoNameLength is End - Start,
   sub_atom(RepoUrl, PrefixLength, RepoNameLength, SuffixLength, RepoName).
 
+% True if DependencyIdent starts with '@metamask' and ends with '-controller'
+is_controller(DependencyIdent) :-
+  Prefix = '@metamask/',
+  atom_length(Prefix, PrefixLength),
+  Suffix = '-controller',
+  atom_length(Suffix, SuffixLength),
+  atom_length(DependencyIdent, DependencyIdentLength),
+  sub_atom(DependencyIdent, 0, PrefixLength, After, Prefix),
+  sub_atom(DependencyIdent, Before, SuffixLength, 0, Suffix),
+  Start is DependencyIdentLength - After + 1,
+  End is Before + 1,
+  RepoNameLength is End - Start,
+  sub_atom(DependencyIdent, PrefixLength, RepoNameLength, SuffixLength, RepoName).
```

Comment:
It appears that `RepoNameLength` in this case is different from the one in `repo_name`. Also, `RepoName` isn't used here (it was used in `repo_name` as it's an argument, but not here). Would this still work?

``` suggestion
  ControllerNameLength is End - Start,
  sub_atom(DependencyIdent, PrefixLength, ControllerNameLength, SuffixLength, _).
```